src/app/WriteClient.cpp

```diff
@@ -396,5 +398,16 @@ CHIP_ERROR WriteClient::ProcessAttributeStatusElement(AttributeStatusElement::Pa
     return err;
 }
 
+CHIP_ERROR WriteClientHandle::SendWriteRequest(NodeId aNodeId, FabricIndex aFabricIndex, SecureSessionHandle * apSecureSession)
+{
+    CHIP_ERROR err = mpWriteClient->SendWriteRequest(aNodeId, aFabricIndex, apSecureSession);
+    if (err != CHIP_NO_ERROR)
+    {
+        mpWriteClient->Shutdown();
```

Comment:
A comment may help here for the fact that SendWriteRequest takes ownership of the write lifetime on success. Maybe:

```cpp
if (err == CHIP_NO_ERROR) {
   // On success, the write client will handle lifetime up until success/falure callbacks are invoked
   mpWriteClient = nullptr;  
} else {
   SetWriteClient(nullptr);
}
return err;
```